### PR TITLE
Add a convention for setting the default discriminator

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Add discriminator validation

### CODE REVIEW

**Comment by AndriySvyryd** [comment]
ping

**Comment by lajones** [comment]
:shipit:

File: test/EntityFramework.Core.Tests/Metadata/ModelConventions/ConventionDispatcherTest.cs [hunk 1]
```diff
@@ -50,6 +50,49 @@ public void OnEntityTypeAdded_calls_apply_on_conventions_in_order()
...
+            });
+            conventions.BaseEntityTypeSetConventions.Add(convention.Object);
+
+            var nullConvention = new Mock<IBaseTypeConvention>();
+            nullConvention.Setup(c => c.Apply(It.IsAny<InternalEntityTypeBuilder>(), It.IsAny<EntityType>()))
+                .Returns<InternalEntityTypeBuilder, EntityType>((b, t) =>
+                {
+                    Assert.Null(t);
+                    Assert.Same(entityTypeBuilder, b);
+                return false;
```
Comment (smitpatel) [inline test/EntityFramework.Core.Tests/Metadata/ModelConventions/ConventionDispatcherTest.cs]: nit: tabing

File: test/EntityFramework.Relational.Tests/Metadata/Conventions/Internal/DiscriminatorConventionTest.cs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            entityTypeBuilder.BaseType(baseTypeBuilder.Metadata, ConfigurationSource.Convention);
+
+            Assert.True(new DiscriminatorConvention().Apply(entityTypeBuilder, oldBaseType: null));
+
+            var discriminator = entityTypeBuilder.Metadata.Relational().DiscriminatorProperty;
+
+            Assert.NotNull(discriminator);
+            Assert.Same(discriminator, baseTypeBuilder.Metadata.Relational().DiscriminatorProperty);
+            Assert.Equal("EntityBase", baseTypeBuilder.Metadata.Relational().DiscriminatorValue);
+            Assert.Equal("Entity", entityTypeBuilder.Metadata.Relational().DiscriminatorValue);
```
Comment (smitpatel) [inline test/EntityFramework.Relational.Tests/Metadata/Conventions/Internal/DiscriminatorConventionTest.cs]: nit:could use `typeof().name` as done in tests below.

**Pull request merged**